qemu/tests/watchdog.py

```diff
@@ -282,6 +286,26 @@ def hotplug_unplug_watchdog_device():
                 test.fail("Oops, find watchdog pci, unplug failed")
             logging.info("The WDT remove successfully")
 
+    def watchdog_test_suit():
+        """
+        Run watchdog-test-framework to verify the function of emulated watchdog
+        devices.
+        """
+
+        _watchdog_device_check(test, session, watchdog_device_type)
+        test_lib = params["test_lib"]
+        test_lib_path = os.path.join(data_dir.get_deps_dir(), test_lib)
+        vm.copy_files_to(test_lib_path, "/home")
+        try:
+            session.cmd_output("/home/%s --yes" % os.path.basename(test_lib_path),
+                               timeout=120)
+        except ShellTimeoutError:
+            o = session.get_output().splitlines()[-1]
+            if 27 <= int(o.rstrip("...")) <= 32:
+                _action_check(test, session, watchdog_action)
+            else:
+                test.fail("Watchdog test suit didn't run successfully.")
```

Comment:
What if no `ShellTimeoutError`? Do nothing, include `_action_check()`?

Reply:
Yes, I should consider the situation. If no ShellTimeoutError, the case will fail.